crates/polars-core/src/datatypes/any_value.rs

```diff
@@ -443,46 +451,52 @@ impl<'a> AnyValue<'a> {
         )
     }
 
-    pub fn cast(&self, dtype: &'a DataType) -> PolarsResult<AnyValue<'a>> {
-        macro_rules! cast_to (
+    pub fn strict_cast(&self, dtype: &'a DataType) -> PolarsResult<AnyValue<'a>> {
+        macro_rules! cast_numeric(
```

Comment:
Can we make this a function? I know we already had this, but I don't understand why we use a macro here. As this creates a lot of code bloat, whereas it can be simple function call.

Reply:
@ritchie46 an alternative I described below is to match on the `(self, dtype)` combination. It means making the feature gates a little messier, but the benefit is the code within the macro because a single default arm and only needs to be defined once:

```rust
match (self, dtype) -> {
    (a, b) -> ...
    (c, d) -> ...
    ...
    (_, _) -> <old macro code>
}